db/version_set.cc

```diff
@@ -1737,12 +1738,43 @@ void Version::Get(const ReadOptions& read_options, const LookupKey& k,
         } else if (fp.GetHitFileLevel() >= 2) {
           RecordTick(db_statistics_, GET_HIT_L2_AND_UP);
         }
-        PERF_COUNTER_BY_LEVEL_ADD(user_key_return_count, 1, fp.GetHitFileLevel());
-        return;
+        PERF_COUNTER_BY_LEVEL_ADD(user_key_return_count, 1,
+                                  fp.GetHitFileLevel());
+        if (do_merge) {
+          return;
+        } else {
+          if (merge_context->GetNumOperands() > (unsigned)num_records) {
```

Comment:
Use C++ style casting, and it should be casted to the same type as merge_context->GetNumOperands().